src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,228 @@
+     */
+    public LectureUnitInformationDTO getSplitUnitData(MultipartFile file) {
+
+        try {
+            log.debug("Start preparing information of split units for the file {}", file);
+
+            List<LectureUnitSplitDTO> units = new ArrayList<>();
+
+            Outline unitsInformation = separateIntoUnits(file);
+            Map<Integer, LectureUnitSplit> unitsDocumentMap = unitsInformation.splits;
+            int numberOfPages = unitsInformation.totalPages;
+
+            unitsDocumentMap.forEach((key, value) -> {
+                LectureUnitSplitDTO newLectureUnit = new LectureUnitSplitDTO(value.unitName, ZonedDateTime.now(), value.startPage, value.endPage);
+                units.add(newLectureUnit);
+            });
+            // return units information, maximum number of pages and by default remove break slides is false
+            return new LectureUnitInformationDTO(units, numberOfPages, false);
+        }
+        catch (IOException e) {
+            log.error("Error while preparing the map with information", e);
+            throw new InternalServerErrorException("Could not prepare split information");
+        }
+    }
+
+    /**
+     * This method prepares a map with information on how the slide
+     * is going to be split. The map looks like the following:
+     * Map<OutlineNumber, (UnitName, StartPage, EndPage)>
+     *
+     * @param file The file (lecture pdf) to be split
+     * @return The prepared map
+     */
+    private Outline separateIntoUnits(MultipartFile file) throws IOException {
+        try (PDDocument document = PDDocument.load(file.getBytes())) {
+            Map<Integer, LectureUnitSplit> outlineMap = new HashMap<>();
+            Splitter pdfSplitter = new Splitter();
+            PDFTextStripper pdfStripper = new PDFTextStripper();
+            // split the document into single pages
+            List<PDDocument> pages = pdfSplitter.split(document);
+            int numberOfPages = document.getNumberOfPages();
+            ListIterator<PDDocument> iterator = pages.listIterator();
+
+            int outlineCount = 0;
+            int index = 1;
+            while (iterator.hasNext()) {
+                PDDocument currentPage = iterator.next();
+                String slideText = pdfStripper.getText(currentPage);
+
+                if (isOutlineSlide(slideText)) {
+                    outlineCount++;
+                    String[] lines = slideText.split("\r\n|\r|\n");
+
+                    // if it's the outline slide it will get the next bullet point as unit name.
+                    String unitName = lines[outlineCount + 1].replaceAll("[^a-zA-Z0-9\\s()_-]", "").replaceFirst("^\\s*", "");
+                    outlineMap.put(outlineCount, new LectureUnitSplit(unitName, outlineCount == 1 ? 1 : index, numberOfPages));
+
+                    if (outlineCount > 1) {
+                        updatePreviousUnitEndPage(outlineCount, outlineMap, index);
+                    }
```

Comment:
```suggestion
                    updatePreviousUnitEndPage(outlineCount, outlineMap, index);
```
Nitpick, preference: moving the check to the helper method simplifies this larger method.